### DISCUSSION TITLE
Proof of concept: Integrate search bar into header

### DISCUSSION THREAD

**Discussion by ransome1** [ransome1/sleek#513] [Show and tell]
Hi gang. 

I want to discuss something, that came up as a feature request. It was about the search bar being perceived as prominent. And yes, it does occupy some space in sleek 1.3.1. Plus you cannot hide it if you don't need it:

![Bildschirmfoto 2023-08-31 um 17 45 13](https://github.com/ransome1/sleek/assets/11188741/c2b6cb7a-478b-4b05-8e09-a4dd69744a14)

While rewriting sleek, I introduced a change (https://github.com/ransome1/sleek/releases/tag/v2.0.0-dev6) to tackle this problem. It is basically to hide the search bar in the header:

![Bildschirmfoto 2023-08-31 um 17 46 41](https://github.com/ransome1/sleek/assets/11188741/1c529cff-94fc-4e11-afff-9267e0f75eb1)

A user can click on the search button or use Cmd/Ctrl + F to open the search bar. Which results in this:

![Bildschirmfoto 2023-08-31 um 17 47 55](https://github.com/ransome1/sleek/assets/11188741/7834eca8-e3fa-4fbe-8a2d-57b3c008bdbf)

Another way of doing this is to integrate it a little bit more with the header, which is now only being used for the file tabs. I created a proof of concept, where the search bar, when triggered, replaces the file tabs. In my opinion this is fine, as the file tabs is not what you need while searching. It could look something like this:

![Bildschirmfoto 2023-08-31 um 17 51 43](https://github.com/ransome1/sleek/assets/11188741/6d840918-480b-459b-83f9-ecb477cd8fe6)

Of course the additional function like "Add as a todo" and the X to remove the search string, would be available:

![Bildschirmfoto 2023-08-31 um 17 52 39](https://github.com/ransome1/sleek/assets/11188741/770b1a82-d292-4611-be1b-40f80237552d)

If you like, you can try out all these proposals yourself. The one, that includes the proof of concept: https://github.com/ransome1/sleek/releases/tag/v2.0.0-dev7

Hope hearing from you guys.

**Comment by CPUGPU** [comment]
I think normally hiding the search bar is a plus, as long as hitting "f" key will still bring it up and give it focus.  I'm not too concerned about what it looks like when called-up.  One school of thought in good UI design is that things shouldn't shift too much when functions like search or selection are brought up (_looking at you_, google voice... :rage:) so I think replacing the tabs bar is better than shifting all the tasks down. Maybe just using a dynamic portion right of the tabs would be better than using all of it?  Don't know how hard that would be to code or if it would really help!

**Comment by fredofredofred** [comment]
Personally I am almost always using the search bar to create new to-dos (or I was in 1.3.1), so even though it is not being used for it's purpose was still useful.

I like the idea of an everything bar which is always there and is used for adding, searching and maybe filtering. 

Maybe a search box could be added to the filter 'side tab' (not sure of terminology) and then people could hide the everything bar if they don't like it without losing functionality.

I agree with the above comment about not having things jump around when something expands. I think your screenshot should faint text in the search box making it clear which file is being searched, I like that solution so people know which file they're searching when they start typing.
